hypothesis-python/src/hypothesis/searchstrategy/regex.py

```diff
@@ -106,6 +106,15 @@ def __init__(self, groups=None, flags=0):
         self.flags = flags
 
 
+def swapcase_matches(c):
+    assert len(c) == 1
+    try:
+        return re.match(c, c.swapcase(), re.IGNORECASE) is not None
+    except re.error:
+        # Issue #1786; a character may be an invalid regex pattern (e.g. r"\")
+        return False
```

Comment:
I don't think this looks like the right fix. If the problem is that we're interpreting a character this is not a valid regex as if it were a valid regex, shouldn't we just be wrapping `c` in a `re.escape`?

Reply:
I don't think `c != c.swapcase()` does the right thing at all does it? It will always be true or there's no harm adding it.

The complicating factor I think is when `len(c.swapcase()) > 1`. If it weren't for that we could just unconditionally add `c.swapcase()`. My concern is there might be other unicode mess we're missing. I'd prefer to use the full check and if performance is a concern we can always cache it somewhere.